### PR TITLE
molten-vk 1.0.15 (new formula)

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

File: Formula/molten-vk.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  depends_on "cmake" => :build
+  depends_on :macos => :sierra # requires IOSurface/IOSurfaceRef.h
+
+  # MoltenVK depends on very specific revisions of its dependencies.
+  # For each resource the path to the file describing the expected
+  # revision is listed.
+  resource "cereal" do
+    # ExternalRevisions/cereal_repo_revision
+    url "https://github.com/USCiLab/cereal/archive/51cbda5f30e56c801c07fe3d3aba5d7fb9e6cca4.tar.gz"
```
Comment (ilovezfs) [inline Formula/molten-vk.rb]: should be
```
url "https://github.com/USCiLab/cereal.git",
    :revision => "51cbda5f30e56c801c07fe3d3aba5d7fb9e6cca4"
```
and the same for the others

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/molten-vk.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  def install
+    (buildpath/"External/cereal").install resource("cereal")
+    (buildpath/"External/glslang").install resource("glslang")
+    (buildpath/"External/glslang/External/spirv-tools").install resource("spirv-tools")
+    ln_s "../extension_enum.inc", buildpath/"External/glslang/External/spirv-tools/source"
+    (buildpath/"External/glslang/External/spirv-tools/external/SPIRV-Headers").install resource("spirv-headers")
+    (buildpath/"External/SPIRV-Cross").install resource("spirv-cross")
+    (buildpath/"External/Vulkan-Headers").install resource("vulkan-headers")
+
+    cd buildpath/"External/Vulkan-Headers" do
```
Comment (ilovezfs) [inline Formula/molten-vk.rb]: only a relative path is needed

File: Formula/molten-vk.rb (hunk 1 above)
Comment (dholm) [reply Formula/molten-vk.rb]: Thanks, fixed!

File: Formula/molten-vk.rb (hunk 2 above)
Comment (dholm) [reply Formula/molten-vk.rb]: Fixed!

File: Formula/molten-vk.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    # External/glslang/known_good.json
+    url "https://github.com/KhronosGroup/SPIRV-Headers/archive/12f8de9f04327336b699b1b80aa390ae7f9ddbf4.tar.gz"
+    sha256 "7d8140b71007d2a7e74903d1b9e66c202c7975d0b3cb60aa1d3af46c9eccfd59"
+  end
+
+  def install
+    (buildpath/"External/cereal").install resource("cereal")
+    (buildpath/"External/glslang").install resource("glslang")
+    (buildpath/"External/glslang/External/spirv-tools").install resource("spirv-tools")
+    ln_s "../extension_enum.inc", buildpath/"External/glslang/External/spirv-tools/source"
```
Comment (ilovezfs) [inline Formula/molten-vk.rb]: This should use `install_symlink`

File: Formula/molten-vk.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      system "make", "install"
+    end
+
+    mkdir "External/glslang/build" do
+      # Required due to files being generated during build.
+      system "cmake", "..", *std_cmake_args
+      system "make", "-C", "External/spirv-tools/source"
+    end
+
+    %w[-macOS GLSLToSPIRVConverter-macOS ShaderConverter SPIRVToMSLConverter-macOS].each do |s|
```
Comment (ilovezfs) [inline Formula/molten-vk.rb]: Where is this list coming from?

File: Formula/molten-vk.rb (hunk 3 above)
Comment (dholm) [reply Formula/molten-vk.rb]: I think this symlink isn't needed anymore, verifying and removing.

File: Formula/molten-vk.rb (hunk 4 above)
Comment (dholm) [reply Formula/molten-vk.rb]: `xcodebuild -list`, I'll add a comment.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Please request a `make install` target upstream. The formula really should be dramatically simpler and could be if the upstream project used git submodules for the resources, and allowed us to just run
```
  def install
    system "cmake", ".", *std_cmake_args
    system "make", "install"
  end
```

**Comment by dholm** [comment]
@ilovezfs They used to be submodules but they removed that in https://github.com/KhronosGroup/MoltenVK/commit/00fda690cae250294a6bb5d2768ba850909aa395.

**Comment by ilovezfs** [comment]
@dholm That's unfortunate. Regardless, it still should use `make install` instead of expecting us to determine what needs to be installed and where it needs to be installed, and expecting us to do those installations manually.

File: Formula/molten-vk.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      system "make", "install"
+    end
+
+    mkdir "External/glslang/build" do
+      # Required due to files being generated during build.
+      system "cmake", "..", *std_cmake_args
+      system "make", "-C", "External/spirv-tools/source"
+    end
+
+    # `macOS` scehemes acquired with `xcodebuild -list`.
```
Comment (Midar) [inline Formula/molten-vk.rb]: Typo

**Comment by Luke-Nukem** [comment]
Been lo0king at doing this. Glad to see it's already handled, though I can't get a successful build on Mojave.

Any chance of an update to most recent then merge of successful build?

File: Formula/molten-vk.rb (hunk 5 above)
Comment (dholm) [reply Formula/molten-vk.rb]: Thanks, fixed.

**Comment by Luke-Nukem** [comment]
Here's the build log I get (with old + new):

```
luke@stardust:~/Downloads/dxvk-0.70$ cat  /Users/luke/Library/Logs/Homebrew/molten-vk/05.xcodebuild
2018-08-30 12:55:05 +1200

xcodebuild
-project
MoltenVKPackaging.xcodeproj
-scheme
MoltenVK-macOS
-configuration
Release
SYMROOT=build
OBJROOT=build

Build settings from command line:
    OBJROOT = build
    SYMROOT = build

2018-08-30 12:55:06.683 xcodebuild[46842:962940] [MT] DVTAssertions: ASSERTION FAILURE in /Library/Caches/com.apple.xbs/Sources/DVTFrameworks/DVTFrameworks-14273.16/DVTFoundation/FilePaths/DVTFilePath.m:908
Details:  Path must be absolute but is not: build
Object:   <DVTFilePath>
Method:   +filePathForPathString:
Thread:   <NSThread: 0x7fced5c18ad0>{number = 1, name = main}
Hints: 

Backtrace:
  0   -[DVTAssertionHandler handleFailureInMethod:object:fileName:lineNumber:assertionSignature:messageFormat:arguments:] (in DVTFoundation)
  1   _DVTAssertionHandler (in DVTFoundation)
  2   _DVTAssertionFailureHandler (in DVTFoundation)
  3   +[DVTFilePath filePathForPathString:] (in DVTFoundation)
  4   -[IDEWorkspace(IDEPIFGenerationCategories) xcbuildIntermediatesPath] (in XCBuildSupport)
  5   IDEXCBuildServiceSession.init(service:for:) (in XCBuildSupport)
  6   IDEXCBuildServiceSession.__allocating_init(service:for:) (in XCBuildSupport)
  7   closure #1 in IDEXCBuildService.serviceSession(for:) (in XCBuildSupport)
  8   partial apply for closure #1 in IDEXCBuildService.serviceSession(for:) (in XCBuildSupport)
  9   closure #1 in IDEXCBuildService.serviceSession(for:)partial apply (in XCBuildSupport)
 10   partial apply for thunk for @callee_guaranteed () -> (@owned IDEXCBuildServiceSession, @error @owned Error) (in XCBuildSupport)
 11   thunk for @callee_guaranteed () -> (@owned IDEXCBuildServiceSession, @error @owned Error)partial apply (in XCBuildSupport)
 12   partial apply for thunk for @callee_guaranteed () -> (@out A, @error @owned Error) (in libswiftDispatch.dylib)
 13   closure #1 in closure #1 in OS_dispatch_queue._syncHelper<A>(fn:execute:rescue:) (in libswiftDispatch.dylib)
 14   partial apply for closure #1 in OS_dispatch_queue._syncHelper<A>(fn:flags:execute:rescue:) (in libswiftDispatch.dylib)
 15   partial apply for closure #1 in closure #1 in OS_dispatch_queue._syncHelper<A>(fn:execute:rescue:) (in libswiftDispatch.dylib)
 16   partial apply for thunk for @callee_guaranteed () -> () (in libswiftDispatch.dylib)
 17   thunk for @callee_guaranteed () -> ()partial apply (in libswiftDispatch.dylib)
 18   thunk for @escaping @callee_guaranteed () -> () (in libswiftDispatch.dylib)
 19   _dispatch_client_callout (in libdispatch.dylib)
 20   _dispatch_lane_barrier_sync_invoke_and_complete (in libdispatch.dylib)
 21   @nonobjc OS_dispatch_queue.sync(execute:) (in libswiftDispatch.dylib)
 22   partial apply for @nonobjc  (in libswiftDispatch.dylib)
 23   OS_dispatch_queue._syncHelper<A>(fn:execute:rescue:) (in libswiftDispatch.dylib)
 24   OS_dispatch_queue.sync<A>(execute:) (in libswiftDispatch.dylib)
 25   @objc IDEXCBuildService.serviceSession(for:) (in XCBuildSupport)
 26   -[IDEXCBuildSystemServiceProvider noteBlueprintProviderDidChange:inWorkspace:] (in XCBuildSupport)
 27   -[IDEWorkspace notifyBuildSystemServiceProviderThatBlueprintProviderDidChange:] (in IDEFoundation)
 28   __46-[IDEWorkspace _referencedContainersDidUpdate]_block_invoke_3 (in IDEFoundation)
 29   -[IDEWorkspace invokeChangingValueForKey:fromSet:toSet:block:] (in IDEFoundation)
 30   -[IDEWorkspace _referencedContainersDidUpdate] (in IDEFoundation)
 31   __38-[IDEWorkspace _setupContainerQueries]_block_invoke.886 (in IDEFoundation)
 32   -[NSObject(DVTObservingConvenience) _dvt_newObserverForKeyPath:options:owner:creationBacktrace:withHandlerBlock:] (in DVTFoundation)
 33   -[NSObject(DVTObservingConvenience) dvt_newObserverForKeyPath:options:withHandlerBlock:] (in DVTFoundation)
 34   -[IDEWorkspace _setupContainerQueries] (in IDEFoundation)
 35   -[IDEWorkspace _setupWorkspaceArenaIfNeeded] (in IDEFoundation)
 36   __81-[IDEWorkspace _finishLoadingAsynchronously:shouldUpgradeFromSimpleFilesFocused:]_block_invoke (in IDEFoundation)
 37   __81-[IDEWorkspace _finishLoadingAsynchronously:shouldUpgradeFromSimpleFilesFocused:]_block_invoke.412 (in IDEFoundation)
 38   __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ (in Foundation)
 39   -[NSBlockOperation main] (in Foundation)
 40   -[__NSOperationInternal _start:] (in Foundation)
 41   __NSOQSchedule_f (in Foundation)
 42   _dispatch_block_async_invoke2 (in libdispatch.dylib)
 43   _dispatch_client_callout (in libdispatch.dylib)
 44   _dispatch_main_queue_callback_4CF (in libdispatch.dylib)
 45   __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ (in CoreFoundation)
 46   __CFRunLoopRun (in CoreFoundation)
 47   CFRunLoopRunSpecific (in CoreFoundation)
 48   +[DVTKVOConditionValidator waitForCondition:sourceObject:keyPathAffectingConditionBlock:timeout:] (in DVTFoundation)
 49   -[Xcode3CommandLineBuildTool _resolveInputOptionsWithTimingSection:] (in Xcode3Core)
 50   -[Xcode3CommandLineBuildTool run] (in Xcode3Core)
 51   main (in xcodebuild)
 52   start (in libdyld.dylib)

HOMEBREW_VERSION: 1.7.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: b119591c6963bff320501f52f9029e4862c00e97
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 14e75a3ee9f97ba4abc10e4e7660682174a77229
Core tap last commit: 79 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CACHE: /Users/luke/Library/Caches/Homebrew
CPU: octa-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.1 => /Applications/Xcode-beta.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 9.0.4, 1.6.0_65-b14-468
macOS: 10.14-x86_64
CLT: 10.0.0.0.1.1533942731
CLT headers: 10.0.0.0.1.1533942731
Xcode: 10.0 => /Applications/Xcode-beta.app/Contents/Developer
XQuartz: 2.7.11 => /opt/X11

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j8
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/libxml2:/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14
HOMEBREW_SDKROOT: /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/cmake/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

**Comment by dholm** [comment]
@Luke-Nukem Looks like a bug in `xcodebuild`.  You should report it to Apple.

**Comment by fxcoudert** [comment]
Given no action on this, we'll welcome a pull request for a new version when upstream moves.

**Pull request closed without merging**